content/blog/2021/08/2021-08-14-introducing-junit-external-storage.adoc

```diff
@@ -0,0 +1,147 @@
+
+https://plugins.jenkins.io/ui/search/?labels=database
+
+e.g. you could install the plugin:database-postgresql[PostgreSQL Database] plugin.
+
+We currently support plugin:database-postgresql[PostgreSQL] or plugin:database-mysql[MySQL], but can support others, just create an issue or send a pull request.
+
+=== From Jenkins UI
+
+Navigate to: Manage Jenkins → Configure System → Junit
+
+In the dropdown select 'SQL Database'
+
+image:/images/post-images/2021/junit-external-storage/junit-sql-config-screen.png[JUnit SQL plugin configuration]
+
+Now configure your Database connection details.
+
+Search for 'Global Database' on the same 'Configure System' page.
+
+Select the database implementation you want to use and click 'Test Connection' to verify Jenkins can connect
+
+image:/images/post-images/2021/junit-external-storage/junit-sql-config-screen.png[JUnit SQL plugin database configuration]
+
+Click 'Save'
+
+=== Configuration as code
+
+If you want to configure the plugin via plugin:configuration-as-code[Configuration as Code] then see the below sample:
+
+[source,yaml]
+----
+unclassified:
+  globalDatabaseConfiguration:
+    database:
+      postgreSQL:
+        database: "jenkins"
+        hostname: "${DB_HOST_NAME}"
+        password: "${DB_PASSWORD}"
+        username: "${DB_USERNAME}"
+        validationQuery: "SELECT 1"
+  junitTestResultStorage:
+    storage: "database"
+----
+
+=== Using the plugin
+
+Now run some builds, here's an example pipeline configuration to get you started if you're just trying out the plugin:
+
+[source,groovy]
+----
+node {
+  writeFile file: 'x.xml', text: '''
+  <testsuite name='sweet' time='200.0'>
+    <testcase classname='Klazz' name='test1' time='198.0'>
+      <error message='failure'/>
+    </testcase>
+    <testcase classname='Klazz' name='test2' time='2.0'/>
+    <testcase classname='other.Klazz' name='test3'>
+    <skipped message='Not actually run.'/>
+    </testcase>
+  </testsuite>
+  '''
+  junit 'x.xml'
+}
+----
+
+You will see a test result trend appear like below on the builds project page:
+
+image:/images/post-images/2021/junit-external-storage/junit-trend.png[JUnit Trend]
+
+If you check on the controller's file system you will see no `junitResult.xml` for new builds.
+
+If you connect to your database and run:
+
+`SELECT * FROM caseresults;`
+
+You will see a number of test results in the database.
+
+== What happens to existing test results?
+
+Existing test results will stay on disk but will not be loaded.
+
+Currently there is no migration scripts or plugin functionality to do this, if you need it then please raise an issue.
+
+
+== How are test results cleaned up
+
+When a job or build is deleted the related test results are removed.
+
+This is expected to be done as part of a 'Build Discarder'.
+
+If you wish to keep your results longer than this you can disable this feature by enabling:
```

Comment:
FWIW `artifact-manager-s3` not only defaults to ignoring Jenkins build deletion events, but does not offer such an option in the GUI/CasC (only an unsupported system property).

Reply:
I ignored it initially but had a few requests, and people saying they wouldn't use it without some type of cleanup 🤷 , mostly just awaiting more feedback in this area on what people need